pkix/csr.go

```diff
@@ -69,8 +70,26 @@ func ParseAndValidateIPs(ipList string) (res []net.IP, err error) {
 	return
 }
 
+// ParseAndValidateURIs parses a comma-delimited list of URIs into an array of url.URLs
+func ParseAndValidateURIs(uriList string) (res []*url.URL, err error) {
+	if len(uriList) > 0 {
+		uris := strings.Split(uriList, ",")
+		for _, uri := range uris {
+			parsedURI, err := url.Parse(uri)
+			if err != nil {
+				parsedURI = nil
+			}
+			if parsedURI == nil {
+				return nil, fmt.Errorf("Invalid URI: %s", uri)
+			}
```

Comment:
url.Parse allows relative URLs, which don't make sense in this context.  We could do something like this:

```suggestion
			}
			if !parsedURI.IsAbs() {
				return nil, fmt.Errorf("Invalid URI: %s", uri)
			}
```

Reply:
And a test to verify '/' and such are rejected